src/domain/settlement/index.js

```diff
@@ -89,6 +95,7 @@ module.exports = {
     const settlement = await SettlementModel.getById({ settlementId }, enums)
     if (settlement) {
       const settlementWindowsList = await SettlementWindowModel.getBySettlementId({ settlementId }, enums)
+      await prepareSettlementWindowsListWithContents(settlementWindowsList)
```

Comment:
It is generally a bad idea to implement functions with side effects. Please, change to pure function, or iterate windows list here, or modify SettlementWindowModel.getBySettlementId if used only here.

Reply:
Implemented a change as suggested